OpenSim/Common/TableSource.h

```diff
@@ -95,21 +121,52 @@ class TableSource_ : public Component {
         return _table;
     }
 
-    /** Replace the existing TimeSeriesTable_ this TableSource_ currently 
-    holds.                                                                    
+    /** Replace the existing TimeSeriesTable_ that this TableSource_ currently 
+    holds. This operation is not allowed if TableSource_ already contains a
+    value for the property `filename`. In such cases, update the property
+    'filename' using `upd_filename()` to empty string and call this function.
 
-    \throws KeyNotFound If table provided does not have column-labels.        */
+    \throws InvalidCall If property `filename` is set. This call is not allowed
+                        if `filename` property is set.                        
+    \throws KeyNotFound If table provided does not have column labels.        */
     void setTable(const Table& table) {
+        OPENSIM_THROW_IF(!get_filename().empty(),
+                         InvalidCall,
+                         "Property 'filename' is set. Cannot set Table now.");
```

Comment:
I am not sure I agree. I read a Model with a TableSource and it has an associated filename. I run that simulation (with Table read in from file) and decide to change the Table and run it again. I am making a conscious and deliberate action to change it.  I would find the exception surprirsing and the extra step to clear the property  annoying.  If the property is not longer in sync with the table then the Component should clear the property. I am telling it which Table to use!

It is also inconsistent with other Components. If I set the frame for a Joint I don't get an exception that the Joint I am setting is conflict with the property, instead the property is updated to represent what I am setting.